src/reducers/ui.js

```diff
@@ -12,15 +12,23 @@ const { prefs } = require("../utils/prefs");
 import type { Action, panelPositionType } from "../actions/types";
 import type { Record } from "../utils/makeRecord";
 
+type searchVisibilityType = {
+    project: boolean,
+    document: boolean
+};
+
 export type UIState = {
-  searchOn: boolean,
+  searchVisibility: Record<searchVisibilityType>,
```

Comment:
Hmm - lets talk about this API

couple of thoughts: 
* visibility is a bit vague and perhaps unecessary
* file is probably more clear than document - i.e. file search, project search
* nested values project / document could be awkward

Here's my proposal: the store has two fields `fileSearch`, `projectSearch` which could have fields: `active`, `mode`, `modifiers`

This would lend itself to actions like `toggleFileSearch(true)` or `setFileSearchMode('function')` or `setFileSearchModifiers({ caseSensitive: true })`